datasets/rd2/data_recipe_with_rd2.yml

```diff
@@ -0,0 +1,337 @@
+## by this is in Enhanced Recurring Donations format.
+## @author Michael Smith
+## @date 2020-08-15
+## ------------------------------------------------------------------------------------
+
+- include_file: recurring_donation_macros.yml
+- include_file: opportunity_macros.yml
+
+## Create "constants" for GAU's and Campaign to use when creating RD's and Opps
+- object: General_Accounting_Unit__c
+  nickname: theRdGau
+  just_once: True
+  fields:
+    Name: The RD GAU
+    Description__c: ${{fake.paragraph}}
+
+- object: General_Accounting_Unit__c
+  nickname: theOppGau
+  just_once: True
+  fields:
+    Name: The Opp GAU
+    Description__c: ${{fake.paragraph}}
+
+- object: General_Accounting_Unit__c
+  count: 100
+  just_once: True
+  fields:
+    Name: The ${{fake.word}} ${{fake.word}} Program
+
+- object: Campaign
+  nickname: theCampaign
+  just_once: True
+  fields:
+    Name: The ${{fake.word}} Campaign
+    IsActive: True
+
+- macro: ContactName
+  ## Assign a random "fake" value for the following fields on the Contact
+  fields:
+    FirstName:
+      fake: first_name
+    LastName:
+      fake: last_name
+    Email:
+      fake: free_email
+    Phone:
+      fake: phone_number
+
+- macro: ContactAddress
+  fields:
+    MailingStreet:
+      fake: street_address
+    MailingCity:
+      fake: city
+    MailingState:
+      fake: state_abbr
+    MailingPostalCode:
+      fake: zipcode
+
+# For each contact:
+#   - 95% of all Contacts will have an Recurring Donation created (in RD2 format)
+#       - 95% of them are for this Household Contact, the rest have this Contact and an Organization Account
+#       - DateEstablished/EffectiveDate is within the last 2 years
+#       - 5% of them have an Allocation and/or a linked Campaign
+#       - 95% of RD's are Open, the other 5% are Fixed length with 24 installments
+#       - There is a random distribution of Installment Periods, DayOfMonth, etc.
+#   - Create 0 to 5 non-RD related Opportunities for each Contact
+#       - CloseDate between 3 years ago and 6 months in the future (ClosedWon if in the past)
+- macro: ContactWithRecurringDonationAndOpps
+  ## Assign a random "fake" value for the following fields on the Contact
+  ## include: ContactName, ContactAddress
+  fields:
+    FirstName:
+      fake: first_name
+    LastName:
+      fake: last_name
+    Email:
+      fake: free_email
+    Phone:
+      fake: phone_number
+    MailingStreet:
+      fake: street_address
+    MailingCity:
+      fake: city
+    MailingState:
+      fake: state_abbr
+    MailingPostalCode:
+      fake: zipcode
+  friends:
+    - object: npe03__Recurring_Donation__c
+      ## 95% of the Contacts will have RD and 5% will not
+      count:
+        random_choice:
+          - choice:
+              probability: 95%
+              pick: 1
+          - choice:
+              probability: 5%
+              pick: 0
+      ## include: RD_RandomDonor, RD_RandomSchedule
```

Comment:
Did you want to turn this on now that macros within macros are allowed?

Reply:
Definitely. I'll test it out shortly.